### PR TITLE
chore: remove projectMigrationCaseReferences feature flag from codebase

### PR DESCRIPTION
**Pull request by cprieststephens**
## Describe your changes

2399: Feature Flag removal - PROJECT_MIGRATION_CASE_REFERENCES
https://pins-ds.atlassian.net/jira/software/c/projects/ASB/boards/166?selectedIssue=ASB-2399

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/forms-web-app/src/pages/projects/_middleware/middleware.component.test.js [hunk 1]
```diff
@@ -13,10 +13,7 @@ jest.mock('../../../config', () => {
 	const originalConfig = jest.requireActual('../../../config');
 	return {
 		...originalConfig,
-		featureFlag: {
-			...originalConfig.featureFlag,
-			projectMigrationCaseReferences: ['mock-case-ref-in-config']
-		}
+		featureFlag: {}
```
Comment (emil-placheta) [inline packages/forms-web-app/src/pages/projects/_middleware/middleware.component.test.js:16]: Why is the `featureFlag` set to an empty object?

Comment (emil-placheta) [reply packages/forms-web-app/src/pages/projects/_middleware/middleware.component.test.js:16]: Looks like config is removed entirely from packages/forms-web-app/src/pages/projects/_middleware/middleware.js

**Review by emil-placheta — CHANGES_REQUESTED** [review]
small comment

File: packages/forms-web-app/src/pages/projects/_middleware/middleware.component.test.js (hunk 1 above)
Comment (cprieststephens) [reply packages/forms-web-app/src/pages/projects/_middleware/middleware.component.test.js]: Thank you, good spot. I've now removed config from the middleware test.

**Review by Philip-Philippou1 — APPROVED** [review]

**Review by emil-placheta — APPROVED** [review]

**Pull request merged**